### PR TITLE
[Translation][Form] Do not translate form labels and placeholders when 'translation_domain' is false

### PR DESCRIPTION
**Pull request by Restless-ET**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Seems like this behaviour was already partially added (as stated [here](https://github.com/symfony/symfony/pull/16016#issuecomment-144464875)) to solve the problem of expanded choice widgets.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig [hunk 1]
```diff
@@ -57,7 +57,7 @@
     {%- endif -%}
     <select {{ block('widget_attributes') }}{% if multiple %} multiple="multiple"{% endif %}>
         {%- if placeholder is not none -%}
-            <option value=""{% if required and value is empty %} selected="selected"{% endif %}>{{ placeholder != '' ? placeholder|trans({}, translation_domain) }}</option>
+            <option value=""{% if required and value is empty %} selected="selected"{% endif %}>{{ placeholder != '' ? (translation_domain is same as (false) ? placeholder : placeholder|trans({}, translation_domain)) }}</option>
```
Comment (aitboudad) [inline src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: please remove the extra space between `as` and `(`
```twig
same as(false)
```

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig [hunk 2]
```diff
@@ -324,7 +324,7 @@
     {%- for attrname, attrvalue in attr -%}
         {{- " " -}}
         {%- if attrname in ['placeholder', 'title'] -%}
-            {{- attrname }}="{{ attrvalue|trans({}, translation_domain) }}"
+            {{- attrname }}="{{ translation_domain is same as (false) ? attrvalue : attrvalue|trans({}, translation_domain) }}"
```
Comment (aitboudad) [inline src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: same here

**Comment by fabpot** [comment]
@Restless-ET Can you add some unit tests to cover the changes?

**Comment by Restless-ET** [comment]
@fabpot Added some unit tests as requested.
Sorry for the delay on this.

**Comment by Tobion** [comment]
:+1: 

Status: Reviewed

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @Restless-ET.

**Pull request closed without merging**